### DISCUSSION TITLE
How scalable do you think manas can / should be?

### DISCUSSION THREAD

**Discussion by argahsuknesib** [manomayam/manas#58] [General]
Hello,
I did some experiments on the amount of concurrent requests the manas server (no authentication, file-based) and around 700 requests (on the same resource) I get this error.
The experiments were done locally (the client, server on the same machine). 
As the server is made in rust, I would expect it to be performant and be able to handle more requests. Do you think of some immediate bottlenecks, which might be causing the problem? 
What do you think can be an achieveable number by the server in future?
![image](https://github.com/manomayam/manas/assets/87450516/69533b8d-4ef8-49af-b7ec-646736293c00)

**Comment by damooo** [comment]
Essentially, it has to make ipc calls to backend. And performance depends on the backend used. If it FS, depends on number of file handles available, etc. We may add a caching layer too in assembled server.  I should soon create proper bench marks to understand proper characterestics with the backend.

Can you run the binary in debug mode, and post full error report?

And till how many requests, it is working fine?

Thank you.
